sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/directconnectivity/rntbd/RntbdClientChannelPool.java

```diff
@@ -1108,7 +1206,7 @@ private void runTasksInPendingAcquisitionQueue() {
 
             task.acquired(true);
             this.acquire(task.promise);
-        }
+        } while (--channelsAvailable> 0);
```

Comment:
Is the move to do-while for readability?

Reply:
No - it is intentionally executing at least once (even if no channels available to allow for more eager connection initiation. Worst case we pull one pendingAcquisition task from the head and re-enqueue it at the tail. Can result in some unfairness - but from test tests we ran and the discussions/analysis in the last couple of days that trade-off is reasonable. Mo's comment above explains the possible unfairness - so I think we are good here?